pkg/diagnostics/tracing.go

```diff
@@ -39,17 +49,43 @@ func SerializeSpanContext(ctx trace.SpanContext) string {
 
 // DeserializeSpanContext deserializes a span context from a string
 func DeserializeSpanContext(ctx string) trace.SpanContext {
+	var ret trace.SpanContext
```

Comment:
Do we still need this ?

Reply:
no. and I removed related codes in similar lines.